browser/src/view/components/Button/Button.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import './Button.scss';
+
+export const Button = ({ layout, label, Icon, type }) => {
+  return (
+    <button
+      className={`${
+        type === ButtonAttributes.Type.destructive
+          ? 'btn-destructive'
+          : type === ButtonAttributes.Type.primary
+          ? 'btn-primary'
+          : 'btn-secondary'
+      }
+      ${
+        type === ButtonAttributes.Layout.iconAndLabel
+          ? 'btn-icon-label'
+          : type === ButtonAttributes.Layout.iconOnly
+          ? 'btn-icon-only'
+          : 'btn-label-only'
+      }`}
+    >
+      {layout === ButtonAttributes.Layout.iconOnly ||
+      layout === ButtonAttributes.Layout.iconAndLabel
+        ? Icon
+        : null}
+      {layout === ButtonAttributes.Layout.labelOnly ||
+      layout === ButtonAttributes.Layout.iconAndLabel
+        ? label
+        : null}
+    </button>
+  );
+};
+
+const ButtonAttributes = {
+  Layout: {
+    iconOnly: 'icon',
+    labelOnly: 'label',
+    iconAndLabel: 'icon-label',
+  },
+  Type: {
+    primary: 'primary',
+    secondary: 'secondary',
+    destructive: 'destructive',
+  },
+};
+Object.freeze(ButtonAttributes);
```

Comment:
This may be overkill.  This should probably be fixed on an organizational level rather than via an Object.freeze.  By organization I mean engineering culture.  The team should know not to change these attributes.